src/shogun/io/SGIO.cpp

```diff
@@ -29,228 +37,188 @@
 
 using namespace shogun;
 
-const EMessageType SGIO::levels[NUM_LOG_LEVELS]={MSG_GCDEBUG, MSG_DEBUG, MSG_INFO, MSG_NOTICE,
-	MSG_WARN, MSG_ERROR, MSG_CRITICAL, MSG_ALERT, MSG_EMERGENCY, MSG_MESSAGEONLY};
-
-const char* SGIO::message_strings[NUM_LOG_LEVELS]={"[GCDEBUG] \0", "[DEBUG] \0", "[INFO] \0",
-	"[NOTICE] \0", "[WARN] \0", "[ERROR] \0",
-	"[CRITICAL] \0", "[ALERT] \0", "[EMERGENCY] \0", "\0"};
-
-const char* SGIO::message_strings_highlighted[NUM_LOG_LEVELS]={"[GCDEBUG] \0", "[DEBUG] \0", "[INFO] \0",
-	"[NOTICE] \0", "\033[1;34m[WARN]\033[0m \0", "\033[1;31m[ERROR]\033[0m \0",
-	"[CRITICAL] \0", "[ALERT] \0", "[EMERGENCY] \0", "\0"};
-
-
-void print_default(FILE* target, const char* str)
+class Formatter : public spdlog::formatter
```

Comment:
This formatter is there just to support the loglevel "MSG_MESSAGEONLY" to not have any patterns